### DISCUSSION TITLE
Need help to configure with a esp-idf project with CLion

### DISCUSSION THREAD

**Discussion by metehus** [slint-ui/slint#5909] [Q&A]
Hello! I'm creating a new project with esp-idf and slint. I got it to build and flash (not fully working yet) via the terminal. But now I'm trying to configure it with CLion but the cmake build fails.
(I'm using Windows 11 btw)
![image](https://github.com/user-attachments/assets/dd1953b1-aad7-4379-acd2-81f5dbadd9d5)
 
I tried to debug the cmake build and these are some information i got:
![image](https://github.com/user-attachments/assets/28551ce4-9db9-448c-97aa-621e36b6177e)

My cmake toolchain is cofigured to use this file as an environment setup:
![image](https://github.com/user-attachments/assets/fa79c9f5-52c6-4b2a-8411-28f3415b6efc)
The contents of `export-esp.bat`:
```bat
@call G:\espressif\frameworks\esp-idf-v5.3-2\export.bat
set PATH=C:\Users\methi\.rustup\toolchains\esp\xtensa-esp-elf\bin;%PATH%
set LIBCLANG_PATH=C:\Users\methi\.rustup\toolchains\esp\xtensa-esp32-elf-clang\esp-clang\bin\libclang.dll
set PATH=C:\Users\methi\.rustup\toolchains\esp\xtensa-esp32-elf-clang\esp-clang\bin;%PATH%
set RUSTUP_TOOLCHAIN=esp
```

The thing I don't understand is that if i run the same bat file and build the project on the command prompt, it works fine:
```
Microsoft Windows [version 10.0.22631.3958]

C:\Users\methi>export-esp.bat
[...]

C:\Users\methi>cd Documents\Projetos\esp32s3-slint-demo

C:\Users\methi\Documents\Projetos\esp32s3-slint-demo>idf.py build
[...]
Project build complete. To flash, run:
 idf.py flash
or
 idf.py -p PORT flash
```

It may be a problem with CLion too, but I'm open for help or suggestions

**Comment by tronical** [comment]
Hmmm, could it be that the environment variables set by `export-esp.bat` aren't set for clion? Is it possible to start clion in a `/devenv` / `/useenv` style like Microsoft Visual Studio, to inherit environment variables from shell?

On the upside, we're working on eliminating the need to run `export-esp.bat` by eliminating the need to have a Rust toolchain installed and offering pre-built Slint binaries instead.

**Reply by metehus** [reply]
update: tried to open on vscode and changed some settings to setup the esp-idf extension, now `slint-esp.h` isn't importing correctly

![image](https://github.com/user-attachments/assets/73dc4eae-a546-4434-8cfc-1becbfad64d8)

**Reply by metehus** [reply]
My friend tried on his machine and he could get it working perfeclty with CLion. I guess it's something on my machine. I will try to reinstall every esp-related tool again

**Reply by metehus** [reply]
well, after reinstalling everything esp related, its working perfectly now!
